test/test_rw_fortran.F90

```diff
@@ -305,27 +306,34 @@ END PROGRAM main
 
 ! Generate a simple, 1D sinusioidal data array with some noise
 SUBROUTINE gen_data(npoints, noise, amp, buf)
-  
   USE ISO_C_BINDING
   IMPLICIT NONE
   INTEGER(C_SIZE_T) :: npoints
   REAL(C_DOUBLE) :: noise
   REAL(C_DOUBLE) :: amp
   REAL(C_DOUBLE), DIMENSION(1:npoints) :: buf
 
+  INTEGER :: size
+  INTEGER, DIMENSION(:), ALLOCATABLE :: seed
   INTEGER(C_SIZE_T) :: i
   REAL(C_DOUBLE) :: x
   REAL(C_DOUBLE) :: pi
   REAL(C_DOUBLE) :: rand
-  pi = 4.0_C_DOUBLE*ATAN(1.0_C_DOUBLE)
 
+  CALL RANDOM_SEED(SIZE=size)
```

Comment:
@jwsblokland I completely agree that using a *fixed* seed is better.

Please pardon my Fortran ignorance (its been many years since I've coded much in Fortran) but I am wondering why having a fixed seed then requires `ALLOCATABLE`? Why not something like `integer, parameter :: seed = 123456789` in the function variable declaration block?

For that matter, @brtnfld now that I am here thinking about it, would you object to `real, parameter :: pi = 3.14159265358979` instead of what you are doing with basically a *derived* formula?

Reply:
The formula adjusts the maximum precision tailored to the architecture but making it a constant also works provided you have enough digits for the precision (and some extra for peace of mind).

I also think it is better to not have a fixed seed since I would rather test for possible extremes and account for them in the test.

Reply:
> The formula adjusts the maximum precision tailored to the architecture but making it a constant also works provided you have enough digits for the precision (and some extra for peace of mind).

Ok, so we're talking at most double precision here, right? So, 16 digits is plenty for that, right? [This ref](http://www.mimirgames.com/articles/programming/digits-of-pi-needed-for-floating-point-numbers/) offers some of the constant values. If we had quad precision, we'd need ~33 digits. This 40 digit value is probably fine... `3.1415926535897932384626433832795028841971`

> I also think it is better to not have a fixed seed since I would rather test for possible extremes and account for them in the test.

I am not sure I follow. If we believe the test is a good test regardless of seed, I think it should be repeatable and so use a fixed seed. What *more* does not having a fixed seed do for us?

Reply:
Oddly, the test fails for certain values; to me, the test should not depend on the random numbers generated whether it fails or passes.  But I think that was because the tolerance in the values check was too extreme. I think this has been addressed, so it should be ok to use a fixed seed. The Fortran test is mainly for testing the wrappers anyway.

Reply:
@markcmiller86 Regarding the allocatable. This is how the subroutine random_seed() has been implemented (see [gfortran random_seed](https://gcc.gnu.org/onlinedocs/gfortran/RANDOM_005fSEED.html) for more details). The be honest, I was also expected one seed value.